back/app/models/static_page.rb

```diff
@@ -20,30 +37,70 @@
 class StaticPage < ApplicationRecord
   CODES = %w[about terms-and-conditions privacy-policy faq proposals custom].freeze
 
+  has_many :pins, as: :page, inverse_of: :page, dependent: :destroy
+  has_many :pinned_admin_publications, through: :pins, source: :admin_publication
   has_one :nav_bar_item, dependent: :destroy
   has_many :static_page_files, -> { order(:ordering) }, dependent: :destroy
   has_many :text_images, as: :imageable, dependent: :destroy
   accepts_nested_attributes_for :nav_bar_item
   accepts_nested_attributes_for :text_images
 
+  before_validation :set_code, on: :create
+  before_validation :generate_slug, on: :create
+
+  before_validation :strip_title
+  before_validation :sanitize_top_info_section_multiloc
+  before_validation :sanitize_bottom_info_section_multiloc
+
+  before_destroy :confirm_is_custom, prepend: true
+
   validates :title_multiloc, presence: true, multiloc: { presence: true }
-  validates :body_multiloc, presence: true, multiloc: { presence: true, html: true }
   validates :slug, presence: true, uniqueness: true
   validates :code, inclusion: { in: CODES }
-  validates :code, uniqueness: true, if: ->(page) { !page.custom? }
+  validates :code, uniqueness: true, unless: :custom?
 
-  before_validation :set_code, on: :create
-  before_validation :generate_slug, on: :create
+  validates :banner_enabled, inclusion: [true, false]
+  validates :banner_layout, inclusion: %w[full_width_banner_layout two_column_layout two_row_layout]
+  validates :banner_overlay_color, css_color: true
+  validates :banner_overlay_opacity, numericality: { only_integer: true,
+                                                     in: [0..100],
+                                                     allow_nil: true }
+  validates :banner_header_multiloc, multiloc: true
+  validates :banner_subheader_multiloc, multiloc: true
+  validates :banner_cta_button_type, inclusion: %w[customized_button no_button]
```

Comment:
Same question here. Are we sure this is a setting that is supposed to set for every individual page?

Reply:
See [my answer](https://github.com/CitizenLabDotCo/citizenlab/pull/2587#discussion_r954694978) to similar question.